question21/main.js

```diff
@@ -0,0 +1,156 @@
+  wrapper.appendChild(table).appendChild(getTableHeadElement(tableHeadValue));
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = (tableHeadValue) => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  for (const value of tableHeadValue) {
+    const th = createElementWithClassName("th", "th");
+    th.textContent = value;
+    value === 'ID' && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th);
+  }
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = () => {
+  const tbody = document.querySelector('tbody');
+  const sortButton = document.getElementById('js-sortbtn');
+  const trElement = [...document.querySelectorAll("tbody > tr")];
+  sortButton.addEventListener('click', (e) => {
+    const clickedCellIndex = e.target.parentElement.cellIndex;
+    const nextStatus = switchSortStatus(e.target);
+    sortButton.dataset.status = nextStatus;
+    const sortedRows = sortFunc(e.target,clickedCellIndex,trElement);
+    while (tbody.firstChild) {
+      tbody.removeChild(tbody.firstChild);
+  }
+    sortedRows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  });
+}
+
+const switchSortStatus = (target) => {
+  switch (target.dataset.status) {
+    case 'default':
+      return 'asc';
+    case 'asc':
+      return 'desc';
+    case 'desc':
+      return 'default'; 
+    default:
+      return 'default';
+  }
+}
+
+const sortFunc = (target,clickedCellIndex,defaultRows) => {
+  const defaultTrElement = [...defaultRows];
+  const currentStatus = target.dataset.status;
+  if (currentStatus === "default") {
+      return defaultRows;
+  }
+  if (currentStatus === "asc"){
+      return defaultTrElement.sort((a,b) => a.children[clickedCellIndex].textContent - b.children[clickedCellIndex].textContent);
+  } 
+  if (currentStatus === "desc"){
```

Comment:
IMO
I think it would reduce the number of unintended bugs if we do the same here and refer to the object.
```suggestion
  if (currentStatus === Sort.Default) {
      return defaultRows;
  }
  if (currentStatus === Sort.Asc){
      return defaultTrElement.sort((a,b) => a.children[clickedCellIndex].textContent - b.children[clickedCellIndex].textContent);
  } 
  if (currentStatus === Sort.Desc){
```

Reply:
This section has also been corrected. Please check it.
fixed commit : https://github.com/csakyo/homework/pull/36/commits/07329f0d8d781c4b305098c3ed97ee981665910b